internal/icinga2/client_api.go

```diff
@@ -322,6 +322,7 @@ func (client *Client) checkMissedChanges(ctx context.Context, objType string, ca
 		case catchupEventCh <- &catchupEventMsg{eventMsg: &eventMsg{ev, attrs.LastStateChange.Time()}}:
 			stateChangeEvents++
 			if fakeEv.Type == event.TypeAcknowledgementSet {
+				fakeEv.ID = 0 // We don't want to reference a previously sent event, so reset its ID!
```

Comment:
Instead of doing this, I think it would be much safer to duplicate the whole event before sending it to any channel. That would be safe without having to argue about whether the first event is safe to be used again at this point. It might be safe due to the channel not being buffered, the receiver processing the events serially, **and** another event being sent in between, but change any of this, and resetting `fakeEv.ID` here would become unsafe.